Allow user to set rabbitmqctl path

Anonymouslemming opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->

Fixes #22945

This replaces https://github.com/ansible/ansible/pull/25601 which I damaged
through a rebase.

Added extra_ctl_paths as an optional setting to allow users who need to run
rabbitmqctl commands as a non-root user to set the path to this binary.

This is only needed by users in non-standard RabbitMQ environments as the
rabbitmqctl script includes checks on which user is running it and fails if
being run by a user other than root or rabbitmqctl.

As an example, on RHEL 6 with RabbitMQ 3.6.2,
/usr/lib/rabbitmq/lib/rabbitmq_server-3.6.2/sbin/rabbitmqctl can be run as a
user other than root or rabbitmq.

This solves the issue for some organisations who do not allow generic service
account names in different environments, or where Rabbit must be run as a
specific user for any other reason

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
messaging module (rabbitmq components)

##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
ansible 2.4.0 (issues_22945_rabbitctl_path_config_test 1c40631382) last updated 2017/07/24 13:03:40 (GMT +100)
  config file = None
  configured module search path = [u'/Users/wpascoe/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/wpascoe/development/ansible_upstream/lib/ansible
  executable location = /Users/wpascoe/development/ansible_upstream/bin/ansible
  python version = 2.7.10 (default, Feb  7 2017, 00:08:15) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)]
```

##### ADDITIONAL INFORMATION
Before the change, attempting to create a rabbit user when rabbitmq is not running as root or rabbitmq would result in the following:

```
TASK [rabbit : add rabbitmq user] ********************************************************************************************************************
fatal: [el6bt]: FAILED! => {"changed": false, "cmd": "/usr/sbin/rabbitmqctl -q list_users", "failed": true, "msg": "Error: could not recognise command", "rc": 1, "stderr": "Error: could not recognise command\n", "stderr_lines": ["Error: could not recognise command"], "stdout": "Usage:\nrabbitmqctl [-n <node>] [-t <timeout>] [-q] <command> [<command options>] \n\nOptions:\n    -n node\n    -q\n    -t timeout\n\nDefault node is \"rabbit@server\", where server is the local host. On a host \nnamed \"server.example.com\", the node name of the RabbitMQ Erlang node will \nusually be rabbit@server (unless RABBITMQ_NODENAME has been set to some \nnon-default value at broker startup time). The output of hostname -s is usually \nthe correct suffix to use after the \"@\" sign. See rabbitmq-server(1) for \ndetails of configuring the RabbitMQ broker.\n\nQuiet output mode is selected with the \"-q\" flag. Informational messages are \nsuppressed when quiet mode is in effect.\n\nOperation timeout in seconds. Only applicable to \"list\" commands. Default is \n\"infinity\".\n\nCommands:\n    stop [<pid_file>]\n    stop_app\n    start_app\n    wait <pid_file>\n    reset\n    force_reset\n    rotate_logs <suffix>\n\n    join_cluster <clusternode> [--ram]\n    cluster_status\n    change_cluster_node_type disc | ram\n    forget_cluster_node [--offline]\n    rename_cluster_node oldnode1 newnode1 [oldnode2] [newnode2 ...]\n    update_cluster_nodes clusternode\n    force_boot\n    sync_queue [-p <vhost>] queue\n    cancel_sync_queue [-p <vhost>] queue\n    purge_queue [-p <vhost>] queue\n    set_cluster_name name\n\n    add_user <username> <password>\n    delete_user <username>\n    change_password <username> <newpassword>\n    clear_password <username>\n    \n            authenticate_user <username> <password>\n          \n    set_user_tags <username> <tag> ...\n    list_users\n\n    add_vhost <vhost>\n    delete_vhost <vhost>\n    list_vhosts [<vhostinfoitem> ...]\n    set_permissions [-p <vhost>] <user> <conf> <write> <read>\n    clear_permissions [-p <vhost>] <username>\n    list_permissions [-p <vhost>]\n    list_user_permissions <username>\n\n    set_parameter [-p <vhost>] <component_name> <name> <value>\n    clear_parameter [-p <vhost>] <component_name> <key>\n    list_parameters [-p <vhost>]\n\n    set_policy [-p <vhost>] [--priority <priority>] [--apply-to <apply-to>] \n<name> <pattern>  <definition>\n    clear_policy [-p <vhost>] <name>\n    list_policies [-p <vhost>]\n\n    list_queues [-p <vhost>] [<queueinfoitem> ...]\n    list_exchanges [-p <vhost>] [<exchangeinfoitem> ...]\n    list_bindings [-p <vhost>] [<bindinginfoitem> ...]\n    list_connections [<connectioninfoitem> ...]\n    list_channels [<channelinfoitem> ...]\n    list_consumers [-p <vhost>]\n    status\n    node_health_check\n    environment\n    report\n    eval <expr>\n\n    close_connection <connectionpid> <explanation>\n    trace_on [-p <vhost>]\n    trace_off [-p <vhost>]\n    set_vm_memory_high_watermark <fraction>\n    set_vm_memory_high_watermark absolute <memory_limit>\n    set_disk_free_limit <disk_limit>\n    set_disk_free_limit mem_relative <fraction>\n\n<vhostinfoitem> must be a member of the list [name, tracing].\n\nThe list_queues, list_exchanges and list_bindings commands accept an optional \nvirtual host parameter for which to display results. The default value is \"/\".\n\n<queueinfoitem> must be a member of the list [name, durable, auto_delete, \narguments, policy, pid, owner_pid, exclusive, exclusive_consumer_pid, \nexclusive_consumer_tag, messages_ready, messages_unacknowledged, messages, \nmessages_ready_ram, messages_unacknowledged_ram, messages_ram, \nmessages_persistent, message_bytes, message_bytes_ready, \nmessage_bytes_unacknowledged, message_bytes_ram, message_bytes_persistent, \nhead_message_timestamp, disk_reads, disk_writes, consumers, \nconsumer_utilisation, memory, slave_pids, synchronised_slave_pids, state].\n\n<exchangeinfoitem> must be a member of the list [name, type, durable, \nauto_delete, internal, arguments, policy].\n\n<bindinginfoitem> must be a member of the list [source_name, source_kind, \ndestination_name, destination_kind, routing_key, arguments].\n\n<connectioninfoitem> must be a member of the list [pid, name, port, host, \npeer_port, peer_host, ssl, ssl_protocol, ssl_key_exchange, ssl_cipher, \nssl_hash, peer_cert_subject, peer_cert_issuer, peer_cert_validity, state, \nchannels, protocol, auth_mechanism, user, vhost, timeout, frame_max, \nchannel_max, client_properties, recv_oct, recv_cnt, send_oct, send_cnt, \nsend_pend, connected_at].\n\n<channelinfoitem> must be a member of the list [pid, connection, name, number, \nuser, vhost, transactional, confirm, consumer_count, messages_unacknowledged, \nmessages_uncommitted, acks_uncommitted, messages_unconfirmed, prefetch_count, \nglobal_prefetch_count].\n\n\n\nOnly root or rabbitmq should run rabbitmqctl\n\n", "stdout_lines": ["Usage:", "rabbitmqctl [-n <node>] [-t <timeout>] [-q] <command> [<command options>] ", "", "Options:", "    -n node", "    -q", "    -t timeout", "", "Default node is \"rabbit@server\", where server is the local host. On a host ", "named \"server.example.com\", the node name of the RabbitMQ Erlang node will ", "usually be rabbit@server (unless RABBITMQ_NODENAME has been set to some ", "non-default value at broker startup time). The output of hostname -s is usually ", "the correct suffix to use after the \"@\" sign. See rabbitmq-server(1) for ", "details of configuring the RabbitMQ broker.", "", "Quiet output mode is selected with the \"-q\" flag. Informational messages are ", "suppressed when quiet mode is in effect.", "", "Operation timeout in seconds. Only applicable to \"list\" commands. Default is ", "\"infinity\".", "", "Commands:", "    stop [<pid_file>]", "    stop_app", "    start_app", "    wait <pid_file>", "    reset", "    force_reset", "    rotate_logs <suffix>", "", "    join_cluster <clusternode> [--ram]", "    cluster_status", "    change_cluster_node_type disc | ram", "    forget_cluster_node [--offline]", "    rename_cluster_node oldnode1 newnode1 [oldnode2] [newnode2 ...]", "    update_cluster_nodes clusternode", "    force_boot", "    sync_queue [-p <vhost>] queue", "    cancel_sync_queue [-p <vhost>] queue", "    purge_queue [-p <vhost>] queue", "    set_cluster_name name", "", "    add_user <username> <password>", "    delete_user <username>", "    change_password <username> <newpassword>", "    clear_password <username>", "    ", "            authenticate_user <username> <password>", "          ", "    set_user_tags <username> <tag> ...", "    list_users", "", "    add_vhost <vhost>", "    delete_vhost <vhost>", "    list_vhosts [<vhostinfoitem> ...]", "    set_permissions [-p <vhost>] <user> <conf> <write> <read>", "    clear_permissions [-p <vhost>] <username>", "    list_permissions [-p <vhost>]", "    list_user_permissions <username>", "", "    set_parameter [-p <vhost>] <component_name> <name> <value>", "    clear_parameter [-p <vhost>] <component_name> <key>", "    list_parameters [-p <vhost>]", "", "    set_policy [-p <vhost>] [--priority <priority>] [--apply-to <apply-to>] ", "<name> <pattern>  <definition>", "    clear_policy [-p <vhost>] <name>", "    list_policies [-p <vhost>]", "", "    list_queues [-p <vhost>] [<queueinfoitem> ...]", "    list_exchanges [-p <vhost>] [<exchangeinfoitem> ...]", "    list_bindings [-p <vhost>] [<bindinginfoitem> ...]", "    list_connections [<connectioninfoitem> ...]", "    list_channels [<channelinfoitem> ...]", "    list_consumers [-p <vhost>]", "    status", "    node_health_check", "    environment", "    report", "    eval <expr>", "", "    close_connection <connectionpid> <explanation>", "    trace_on [-p <vhost>]", "    trace_off [-p <vhost>]", "    set_vm_memory_high_watermark <fraction>", "    set_vm_memory_high_watermark absolute <memory_limit>", "    set_disk_free_limit <disk_limit>", "    set_disk_free_limit mem_relative <fraction>", "", "<vhostinfoitem> must be a member of the list [name, tracing].", "", "The list_queues, list_exchanges and list_bindings commands accept an optional ", "virtual host parameter for which to display results. The default value is \"/\".", "", "<queueinfoitem> must be a member of the list [name, durable, auto_delete, ", "arguments, policy, pid, owner_pid, exclusive, exclusive_consumer_pid, ", "exclusive_consumer_tag, messages_ready, messages_unacknowledged, messages, ", "messages_ready_ram, messages_unacknowledged_ram, messages_ram, ", "messages_persistent, message_bytes, message_bytes_ready, ", "message_bytes_unacknowledged, message_bytes_ram, message_bytes_persistent, ", "head_message_timestamp, disk_reads, disk_writes, consumers, ", "consumer_utilisation, memory, slave_pids, synchronised_slave_pids, state].", "", "<exchangeinfoitem> must be a member of the list [name, type, durable, ", "auto_delete, internal, arguments, policy].", "", "<bindinginfoitem> must be a member of the list [source_name, source_kind, ", "destination_name, destination_kind, routing_key, arguments].", "", "<connectioninfoitem> must be a member of the list [pid, name, port, host, ", "peer_port, peer_host, ssl, ssl_protocol, ssl_key_exchange, ssl_cipher, ", "ssl_hash, peer_cert_subject, peer_cert_issuer, peer_cert_validity, state, ", "channels, protocol, auth_mechanism, user, vhost, timeout, frame_max, ", "channel_max, client_properties, recv_oct, recv_cnt, send_oct, send_cnt, ", "send_pend, connected_at].", "", "<channelinfoitem> must be a member of the list [pid, connection, name, number, ", "user, vhost, transactional, confirm, consumer_count, messages_unacknowledged, ", "messages_uncommitted, acks_uncommitted, messages_unconfirmed, prefetch_count, ", "global_prefetch_count].", "", "", "", "Only root or rabbitmq should run rabbitmqctl", ""]}
```

After the change, this succeeds:

```
TASK [rabbit : add rabbitmq user] ********************************************************************************************************************
ok: [host]
```

Commit:
Allow user to set rabbitmqctl path

Fixes #22945

This replaces https://github.com/ansible/ansible/pull/25601 which I damaged
through a rebase.

Added extra_ctl_paths as an optional setting to allow users who need to run
rabbitmqctl commands as a non-root user to set the path to this binary.

This is only needed by users in non-standard RabbitMQ environments as the
rabbitmqctl script includes checks on which user is running it and fails if
being run by a user other than root or rabbitmqctl.

As an example, on RHEL 6 with RabbitMQ 3.6.2,
/usr/lib/rabbitmq/lib/rabbitmq_server-3.6.2/sbin/rabbitmqctl can be run as a
user other than root or rabbitmq.

This solves the issue for some organisations who do not allow generic service
account names in different environments, or where Rabbit must be run as a
specific user for any other reason

*ISSUE TYPE*

* Feature Pull Request

*COMPONENT NAME*

messaging module (rabbitmq components)

*ANSIBLE VERSION*

```
ansible 2.4.0 (issues_22945_rabbitctl_path_config_test 1c40631382) last updated 2017/07/24 13:03:40 (GMT +100)
  config file = None
  configured module search path = [u'/Users/wpascoe/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/wpascoe/development/ansible_upstream/lib/ansible
  executable location = /Users/wpascoe/development/ansible_upstream/bin/ansible
  python version = 2.7.10 (default, Feb  7 2017, 00:08:15) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)]
```

*ADDITIONAL INFORMATION*

Before the change, attempting to create a rabbit user when rabbitmq is not running as root or rabbitmq would result in the following:

```
TASK [rabbit : add rabbitmq user] ********************************************************************************************************************
fatal: [el6bt]: FAILED! => {"changed": false, "cmd": "/usr/sbin/rabbitmqctl -q list_users", "failed": true, "msg": "Error: could not recognise command", "rc": 1, "stderr": "Error: could not recognise command\n", "stderr_lines": ["Error: could not recognise command"], "stdout": "Usage:\nrabbitmqctl [-n <node>] [-t <timeout>] [-q] <command> [<command options>] \n\nOptions:\n    -n node\n    -q\n    -t timeout\n\nDefault node is \"rabbit@server\", where server is the local host. On a host \nnamed \"server.example.com\", the node name of the RabbitMQ Erlang node will \nusually be rabbit@server (unless RABBITMQ_NODENAME has been set to some \nnon-default value at broker startup time). The output of hostname -s is usually \nthe correct suffix to use after the \"@\" sign. See rabbitmq-server(1) for \ndetails of configuring the RabbitMQ broker.\n\nQuiet output mode is selected with the \"-q\" flag. Informational messages are \nsuppressed when quiet mode is in effect.\n\nOperation timeout in seconds. Only applicable to \"list\" commands. Default is \n\"infinity\".\n\nCommands:\n    stop [<pid_file>]\n    stop_app\n    start_app\n    wait <pid_file>\n    reset\n    force_reset\n    rotate_logs <suffix>\n\n    join_cluster <clusternode> [--ram]\n    cluster_status\n    change_cluster_node_type disc | ram\n    forget_cluster_node [--offline]\n    rename_cluster_node oldnode1 newnode1 [oldnode2] [newnode2 ...]\n    update_cluster_nodes clusternode\n    force_boot\n    sync_queue [-p <vhost>] queue\n    cancel_sync_queue [-p <vhost>] queue\n    purge_queue [-p <vhost>] queue\n    set_cluster_name name\n\n    add_user <username> <password>\n    delete_user <username>\n    change_password <username> <newpassword>\n    clear_password <username>\n    \n            authenticate_user <username> <password>\n          \n    set_user_tags <username> <tag> ...\n    list_users\n\n    add_vhost <vhost>\n    delete_vhost <vhost>\n    list_vhosts [<vhostinfoitem> ...]\n    set_permissions [-p <vhost>] <user> <conf> <write> <read>\n    clear_permissions [-p <vhost>] <username>\n    list_permissions [-p <vhost>]\n    list_user_permissions <username>\n\n    set_parameter [-p <vhost>] <component_name> <name> <value>\n    clear_parameter [-p <vhost>] <component_name> <key>\n    list_parameters [-p <vhost>]\n\n    set_policy [-p <vhost>] [--priority <priority>] [--apply-to <apply-to>] \n<name> <pattern>  <definition>\n    clear_policy [-p <vhost>] <name>\n    list_policies [-p <vhost>]\n\n    list_queues [-p <vhost>] [<queueinfoitem> ...]\n    list_exchanges [-p <vhost>] [<exchangeinfoitem> ...]\n    list_bindings [-p <vhost>] [<bindinginfoitem> ...]\n    list_connections [<connectioninfoitem> ...]\n    list_channels [<channelinfoitem> ...]\n    list_consumers [-p <vhost>]\n    status\n    node_health_check\n    environment\n    report\n    eval <expr>\n\n    close_connection <connectionpid> <explanation>\n    trace_on [-p <vhost>]\n    trace_off [-p <vhost>]\n    set_vm_memory_high_watermark <fraction>\n    set_vm_memory_high_watermark absolute <memory_limit>\n    set_disk_free_limit <disk_limit>\n    set_disk_free_limit mem_relative <fraction>\n\n<vhostinfoitem> must be a member of the list [name, tracing].\n\nThe list_queues, list_exchanges and list_bindings commands accept an optional \nvirtual host parameter for which to display results. The default value is \"/\".\n\n<queueinfoitem> must be a member of the list [name, durable, auto_delete, \narguments, policy, pid, owner_pid, exclusive, exclusive_consumer_pid, \nexclusive_consumer_tag, messages_ready, messages_unacknowledged, messages, \nmessages_ready_ram, messages_unacknowledged_ram, messages_ram, \nmessages_persistent, message_bytes, message_bytes_ready, \nmessage_bytes_unacknowledged, message_bytes_ram, message_bytes_persistent, \nhead_message_timestamp, disk_reads, disk_writes, consumers, \nconsumer_utilisation, memory, slave_pids, synchronised_slave_pids, state].\n\n<exchangeinfoitem> must be a member of the list [name, type, durable, \nauto_delete, internal, arguments, policy].\n\n<bindinginfoitem> must be a member of the list [source_name, source_kind, \ndestination_name, destination_kind, routing_key, arguments].\n\n<connectioninfoitem> must be a member of the list [pid, name, port, host, \npeer_port, peer_host, ssl, ssl_protocol, ssl_key_exchange, ssl_cipher, \nssl_hash, peer_cert_subject, peer_cert_issuer, peer_cert_validity, state, \nchannels, protocol, auth_mechanism, user, vhost, timeout, frame_max, \nchannel_max, client_properties, recv_oct, recv_cnt, send_oct, send_cnt, \nsend_pend, connected_at].\n\n<channelinfoitem> must be a member of the list [pid, connection, name, number, \nuser, vhost, transactional, confirm, consumer_count, messages_unacknowledged, \nmessages_uncommitted, acks_uncommitted, messages_unconfirmed, prefetch_count, \nglobal_prefetch_count].\n\n\n\nOnly root or rabbitmq should run rabbitmqctl\n\n", "stdout_lines": ["Usage:", "rabbitmqctl [-n <node>] [-t <timeout>] [-q] <command> [<command options>] ", "", "Options:", "    -n node", "    -q", "    -t timeout", "", "Default node is \"rabbit@server\", where server is the local host. On a host ", "named \"server.example.com\", the node name of the RabbitMQ Erlang node will ", "usually be rabbit@server (unless RABBITMQ_NODENAME has been set to some ", "non-default value at broker startup time). The output of hostname -s is usually ", "the correct suffix to use after the \"@\" sign. See rabbitmq-server(1) for ", "details of configuring the RabbitMQ broker.", "", "Quiet output mode is selected with the \"-q\" flag. Informational messages are ", "suppressed when quiet mode is in effect.", "", "Operation timeout in seconds. Only applicable to \"list\" commands. Default is ", "\"infinity\".", "", "Commands:", "    stop [<pid_file>]", "    stop_app", "    start_app", "    wait <pid_file>", "    reset", "    force_reset", "    rotate_logs <suffix>", "", "    join_cluster <clusternode> [--ram]", "    cluster_status", "    change_cluster_node_type disc | ram", "    forget_cluster_node [--offline]", "    rename_cluster_node oldnode1 newnode1 [oldnode2] [newnode2 ...]", "    update_cluster_nodes clusternode", "    force_boot", "    sync_queue [-p <vhost>] queue", "    cancel_sync_queue [-p <vhost>] queue", "    purge_queue [-p <vhost>] queue", "    set_cluster_name name", "", "    add_user <username> <password>", "    delete_user <username>", "    change_password <username> <newpassword>", "    clear_password <username>", "    ", "            authenticate_user <username> <password>", "          ", "    set_user_tags <username> <tag> ...", "    list_users", "", "    add_vhost <vhost>", "    delete_vhost <vhost>", "    list_vhosts [<vhostinfoitem> ...]", "    set_permissions [-p <vhost>] <user> <conf> <write> <read>", "    clear_permissions [-p <vhost>] <username>", "    list_permissions [-p <vhost>]", "    list_user_permissions <username>", "", "    set_parameter [-p <vhost>] <component_name> <name> <value>", "    clear_parameter [-p <vhost>] <component_name> <key>", "    list_parameters [-p <vhost>]", "", "    set_policy [-p <vhost>] [--priority <priority>] [--apply-to <apply-to>] ", "<name> <pattern>  <definition>", "    clear_policy [-p <vhost>] <name>", "    list_policies [-p <vhost>]", "", "    list_queues [-p <vhost>] [<queueinfoitem> ...]", "    list_exchanges [-p <vhost>] [<exchangeinfoitem> ...]", "    list_bindings [-p <vhost>] [<bindinginfoitem> ...]", "    list_connections [<connectioninfoitem> ...]", "    list_channels [<channelinfoitem> ...]", "    list_consumers [-p <vhost>]", "    status", "    node_health_check", "    environment", "    report", "    eval <expr>", "", "    close_connection <connectionpid> <explanation>", "    trace_on [-p <vhost>]", "    trace_off [-p <vhost>]", "    set_vm_memory_high_watermark <fraction>", "    set_vm_memory_high_watermark absolute <memory_limit>", "    set_disk_free_limit <disk_limit>", "    set_disk_free_limit mem_relative <fraction>", "", "<vhostinfoitem> must be a member of the list [name, tracing].", "", "The list_queues, list_exchanges and list_bindings commands accept an optional ", "virtual host parameter for which to display results. The default value is \"/\".", "", "<queueinfoitem> must be a member of the list [name, durable, auto_delete, ", "arguments, policy, pid, owner_pid, exclusive, exclusive_consumer_pid, ", "exclusive_consumer_tag, messages_ready, messages_unacknowledged, messages, ", "messages_ready_ram, messages_unacknowledged_ram, messages_ram, ", "messages_persistent, message_bytes, message_bytes_ready, ", "message_bytes_unacknowledged, message_bytes_ram, message_bytes_persistent, ", "head_message_timestamp, disk_reads, disk_writes, consumers, ", "consumer_utilisation, memory, slave_pids, synchronised_slave_pids, state].", "", "<exchangeinfoitem> must be a member of the list [name, type, durable, ", "auto_delete, internal, arguments, policy].", "", "<bindinginfoitem> must be a member of the list [source_name, source_kind, ", "destination_name, destination_kind, routing_key, arguments].", "", "<connectioninfoitem> must be a member of the list [pid, name, port, host, ", "peer_port, peer_host, ssl, ssl_protocol, ssl_key_exchange, ssl_cipher, ", "ssl_hash, peer_cert_subject, peer_cert_issuer, peer_cert_validity, state, ", "channels, protocol, auth_mechanism, user, vhost, timeout, frame_max, ", "channel_max, client_properties, recv_oct, recv_cnt, send_oct, send_cnt, ", "send_pend, connected_at].", "", "<channelinfoitem> must be a member of the list [pid, connection, name, number, ", "user, vhost, transactional, confirm, consumer_count, messages_unacknowledged, ", "messages_uncommitted, acks_uncommitted, messages_unconfirmed, prefetch_count, ", "global_prefetch_count].", "", "", "", "Only root or rabbitmq should run rabbitmqctl", ""]}
```

After the change, this succeeds:

```
TASK [rabbit : add rabbitmq user] ********************************************************************************************************************
ok: [host]
```

Anonymouslemming commented:
@chrishoffman @manuel-sousa @romanek-adam

This PR replaces https://github.com/ansible/ansible/pull/25601 with the changes requested by @bcoca made. Please could you let me know if this is an acceptable PR.

Thanks,

hyperized commented:
@Anonymouslemming I usually follow this guide for rebasing: https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request and I've yet to do it wrong that way.

That said, can you rebase :)?

Anonymouslemming commented:
@hyperized I've rebased again (first rebase was about 10 days ago). I didn't see any changes, but I've squashed my commit message as part of this. Please let me know if this looks better ?

Thanks!

Akasurde commented on lib/ansible/utils/module_docs_fragments/rabbitmq.py:
```diff
@@ -0,0 +1,32 @@
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
```
Could you please use - 

```
# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
# Copyright: (c) 2017, Ansible Project
``` 

instead of line 1 to 17

Anonymouslemming replied:
Done :)

Akasurde commented on lib/ansible/utils/module_docs_fragments/rabbitmq.py:
```diff
@@ -0,0 +1,32 @@
+        - Only needed when running RabbitMQ as user other than root / rabbitmq
+      required: false
+      default: ()
+      version_added: "2.4"
```
version_added: "2.5"

Anonymouslemming replied:
Done :)

Akasurde reviewed (changes requested):
Apply requested changes and rest LGTM

WTPascoe committed:
Licence and version changes as requested

WTPascoe committed:
pep8 fix - add blank line

Anonymouslemming commented:
@Akasurde I've made the changes that you're requested - thanks :)

Akasurde commented:
I am OK with changes, but need one ship it.

hyperized commented:
shipit

Anonymouslemming commented:
@Akasurde I see that @hyperized has added a shipit - are you ok to mark your requested change as complete, or is that something I need to do somehow ? 

Thanks,

biomassives commented:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (104/100 (104%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from o1.sgmail.github.com ([192.254.114.176]:47706)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES128-GCM-SHA256:128)
	(Exim 4.89_1)
	(envelope-from <bounces+848413-b37a-greg=biomassiv.es@sgmail.github.com>)
	id 1eWpl6-0034yo-TJ
	for greg@biomassiv.es; Wed, 03 Jan 2018 14:35:32 -0600
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed/relaxed; d=github.com; 
	h=from:reply-to:to:cc:in-reply-to:references:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=PmDoTVXY9G0ZhmXaOHwBgB8lQKo=; b=HfnP9pWBvf+wPEFI
	D32+/nEimfwyQz6DODqZznbVF8GPLR+Hp77ddf4irB8u2gPo3fwSmnOs3FoW4uud
	5qZIZKrEGA7CyvV8tJSFysHb1nq1M6ZoxFaVFe47EsH6koN6B/YXEQvu5bLKECcN
	kmhpeUTlnJIfN4I0XEJ+QN2jtlQ=
Received: by filter1221p1mdw1.sendgrid.net with SMTP id filter1221p1mdw1-13333-5A4D3E6D-14
        2018-01-03 20:34:53.802035819 +0000 UTC
Received: from github-smtp2b-ext-cp1-prd.iad.github.net (github-smtp2b-ext-cp1-prd.iad.github.net [192.30.253.17])
	by ismtpd0005p1iad1.sendgrid.net (SG) with ESMTP id rIFvcIjjRgiSBU1rMKFOsQ
	for <greg@biomassiv.es>; Wed, 03 Jan 2018 20:34:53.779 +0000 (UTC)
Date: Wed, 03 Jan 2018 20:34:53 +0000 (UTC)
From: Wayne <notifications@github.com>
Reply-To: ansible/ansible <reply@reply.github.com>
To: ansible/ansible <ansible@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Message-ID: <ansible/ansible/pull/27263/c355119632@github.com>
In-Reply-To: <ansible/ansible/pull/27263@github.com>
References: <ansible/ansible/pull/27263@github.com>
Subject: Re: [ansible/ansible] Allow user to set rabbitmqctl path (#27263)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_5a4d3e6da9406_f6e3f8765a16f2c395129";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: Anonymouslemming
X-GitHub-Recipient: biomassives
X-GitHub-Reason: subscribed
List-ID: ansible/ansible <ansible.ansible.github.com>
List-Archive: https://github.com/ansible/ansible
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+0042d4e2ea01d810fc2357e341c20214d6c35bb43a1354b092cf000000011665006d92a169ce0ea0094a@reply.github.com>,
 <https://github.com/notifications/unsubscribe/AELU4kO_X57hyC9u05n0UJD-XruoKk00ks5tG-RtgaJpZM4Oiax8>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: greg@biomassiv.es
X-SG-EID: ZlaDthyue/4HG+EgTb7JoZYs48JNb2iQtCcPV6oIs9V9P/JnohmU2GoSr83EF0BbkazxsuxH7pbmoj
 ItgsqY/RRSyjrwJr0li5g9eEMIAueEBBYJoWaTFUYIWGWideznNbPE5MzsGsZMdm8zHaLq+RI3I+Ef
 +dsXQouATCatHJMi04AHyK8E2uLe/ocz0bxXF72bzJ1JLlQroi8kzNWL+A==

----==_mimepart_5a4d3e6da9406_f6e3f8765a16f2c395129
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

@Akasurde I see that @hyperized has added a shipit - are you ok to mark your requested change as complete, or is that something I need to do somehow ? 

Thanks,

-- 
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/ansible/ansible/pull/27263#issuecomment-355119632
----==_mimepart_5a4d3e6da9406_f6e3f8765a16f2c395129
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p><a href="https://github.com/akasurde" class="user-mention">@Akasurde</a> I see that <a href="https://github.com/hyperized" class="user-mention">@hyperized</a> has added a shipit - are you ok to mark your requested change as complete, or is that something I need to do somehow ?</p>
<p>Thanks,</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br />You are receiving this because you are subscribed to this thread.<br />Reply to this email directly, <a href="https://github.com/ansible/ansible/pull/27263#issuecomment-355119632">view it on GitHub</a>, or <a href="https://github.com/notifications/unsubscribe-auth/AELU4lN3046m2Zp1WNthjAf4RYZXFvwlks5tG-RtgaJpZM4Oiax8">mute the thread</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/AELU4n2A-zM4QG_rflaO1kMWT6LjSP51ks5tG-RtgaJpZM4Oiax8.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
<div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
  <link itemprop="url" href="https://github.com/ansible/ansible/pull/27263#issuecomment-355119632"></link>
  <meta itemprop="name" content="View Pull Request"></meta>
</div>
<meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

<script type="application/json" data-scope="inboxmarkup">{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/ansible/ansible","title":"ansible/ansible","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/ansible/ansible"}},"updates":{"snippets":[{"icon":"PERSON","message":"@Anonymouslemming in #27263: @Akasurde I see that @hyperized has added a shipit - are you ok to mark your requested change as complete, or is that something I need to do somehow ? \r\n\r\nThanks,"}],"action":{"name":"View Pull Request","url":"https://github.com/ansible/ansible/pull/27263#issuecomment-355119632"}}}</script>
----==_mimepart_5a4d3e6da9406_f6e3f8765a16f2c395129--

Anonymouslemming commented:
Hi @Akasurde 

Any updates on who I could speak to to move this forward please ? 

Thanks,
